binarytree/__init__.py

```diff
@@ -39,6 +46,15 @@ def _new_node(value):
         return _node_init_func(value)
     return (_node_cls or Node)(value)
 
+def _get_id():
```

Comment:
Instead of using a global array (which will grow and make the `max` operation slower), why not a counter?